frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx

```diff
@@ -478,7 +480,11 @@ export class UnconnectedDataSelector extends Component {
   };
 
   hasSavedQuestions = () => {
-    return this.state.databases.some(database => database.is_saved_questions);
+    const { canSelectSavedQuestion } = this.props;
```

Comment:
Use pretty much the same structure as `hasMetrics`, so they're consistent.
https://github.com/metabase/metabase/blob/08c8ab915d445f0a871fb9803e0d492a6eda34f3/frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx#L468-L471